cmd/soroban-cli/src/events.rs

```diff
@@ -0,0 +1,304 @@
+use clap::{ArgEnum, Parser};
+use termcolor::{Color, ColorChoice, StandardStream, WriteColor};
+use termcolor_output::colored;
+
+use soroban_env_host::xdr::{self, ReadXdr};
+
+use crate::rpc::{Client, Event};
+use crate::{rpc, snapshot, utils};
+use crate::{HEADING_RPC, HEADING_SANDBOX};
+
+#[derive(Parser, Debug)]
+#[clap()]
+pub struct Cmd {
+    /// The first ledger sequence number in the range to pull events
+    #[clap(short, long)]
+    start_ledger: u32,
+
+    /// The last (and inclusive) ledger sequence number in the range to pull events
+    /// https://developers.stellar.org/docs/encyclopedia/ledger-headers#ledger-sequence
+    #[clap(short, long)]
+    end_ledger: u32,
+
+    /// Formatting options for outputted events
+    #[clap(long, arg_enum, default_value = "pretty")]
+    format: OutputFormat,
+
+    /// RPC server endpoint
+    #[clap(long,
+        env = "SOROBAN_RPC_URL",
+        help_heading = HEADING_RPC,
+        conflicts_with = "events-file",
+        conflicts_with = "follow",
+    )]
+    rpc_url: Option<String>,
+
+    /// Local event store (likely generated by `invoke`) to pull events from
+    #[clap(
+        long,
+        parse(from_os_str),
+        value_name = "PATH",
+        env = "SOROBAN_EVENTS_FILE",
+        help_heading = HEADING_SANDBOX,
+        conflicts_with = "rpc-url",
+    )]
+    events_file: Option<std::path::PathBuf>,
```

Comment:
What should we do if the file doesn't exist? error? exit silently? It might not exist because the user put the wrong path. But it might not exist because there are no events yet.. Tricky... 🤔

Reply:
(or if the file is empty)

Reply:
I already handle this with `InvalidFile` if it doesn't exist and `CannotReadFile` if reading fails. I don't think we need to worry about the non-existence case since this is meant to be run after `invoke`.

Maybe if `--follow` is specified, we can skip the non-existence error? As it implies, more heavily at least, that it's running _alongside_ rather than _after_ invoke.

Reply:
I'd either say we should either always skip non-existence error, or always output, it for consistency.

Reply:
Yep, always error.